pygmt/tests/test_basemap.py

```diff
@@ -54,15 +56,17 @@ def test_basemap_power_axis():
     return fig
 
 
-@pytest.mark.xfail(
-    reason="Baseline image not updated to use earth relief grid in GMT 6.1.0",
-)
-@pytest.mark.mpl_image_compare
+@check_figures_equal()
 def test_basemap_polar():
     "Create a polar basemap plot"
-    fig = Figure()
-    fig.basemap(R="0/360/0/1000", J="P6i", B="afg")
-    return fig
+    fig_test = Figure()
+    fig_test.basemap(R="0/360/0/1000", J="P6i", B="afg")
+
+    fig_ref = Figure()
+    with Session() as lib:
+        lib.call_module("basemap", "-R0/360/0/1000 -JP6i -Bafg")
```

Comment:
The reference image is created by directly calling the low-level API function `call_module`.

Reply:
I like the concept in general (testing against the low level GMT API), but this can get complicated real fast for some modules that output to a file (see e.g. @liamtoney's example at https://github.com/GenericMappingTools/pygmt/issues/364#issuecomment-623179721 for `grdgradient`).

Should we have a convenience function to wrap around this low-level API better? E.g. something like `pygmt.gmt(module="basemap", arg_str="-R0/360/0/1000 -JP6i -Bafg") that removes the `with` statement.

Reply:
> but this can get complicated real fast for some modules that output to a file (see e.g. @liamtoney's example at [#364 (comment)](https://github.com/GenericMappingTools/pygmt/issues/364#issuecomment-623179721) for `grdgradient`).

Most tests are simple, testing only one aspect of functions. https://github.com/GenericMappingTools/pygmt/issues/364#issuecomment-623179721's example may be the most  complicated one, but we won't use virtual files for the references images. 

> Should we have a convenience function to wrap around this low-level API better? E.g. something like `pygmt.gmt(module="basemap", arg_str="-R0/360/0/1000 -JP6i -Bafg") that removes the `with` statement.

I believe @leouieda is against it (https://github.com/GenericMappingTools/pygmt/issues/122#issuecomment-358436386).

Reply:
> Most tests are simple, testing only one aspect of functions. [#364 (comment)](https://github.com/GenericMappingTools/pygmt/issues/364#issuecomment-623179721) example may be the most complicated one, but we won't use virtual files for the references images.

True, but as you mentioned, this way of testing isn't very Pythonic or new contributor friendly. We can definitely think of a better solution that doesn't involve `clib` here, just need a bit more time, and maybe have a look around some other projects like `GMT.jl` and `matplotlib` to see how they do it.

> I believe @leouieda is against it ([#122 (comment)](https://github.com/GenericMappingTools/pygmt/issues/122#issuecomment-358436386)).

Ah yes, I thought I saw this somewhere before, thanks for linking. Definitely hold on the `pygmt.gmt`/`fig.rawcommand` feature then.